src/components/util/overlays/popover/usePopover.ts

```diff
@@ -79,6 +79,35 @@ export const usePopover = <E extends HTMLElement>(
     }
   }, [controller.activate, controller.deactivate]);
   
+  // Added for outside click detection
+  React.useEffect(() => {
+  if (!controller.active) return;
+
+  const handleClickOutside = (event: MouseEvent) => {
+    const target = event.target as Node;
+    const popover = popoverRef.current;
+    
+    // Prevent popover from closing if click was inside the popover
+    if (!popover || popover.contains(target)) return;
+
+    // Prevent popover from closing if click was inside portal
+    const isInPortal = (target as HTMLElement).closest('[data-portal]');
```

Comment:
Here also, check to see if `target` is actually an `HTMLElement` (and then you won't need the `Node` check above since `HTMLElement` is a subtype)